types/Connection.d.ts

```diff
@@ -0,0 +1,212 @@
+      readonly db: mongodb.Db;
+
+      /**
+       * Helper for `createCollection()`. Will explicitly create the given collection
+       * with specified options. Used to create [capped collections](https://docs.mongodb.com/manual/core/capped-collections/)
+       * and [views](https://docs.mongodb.com/manual/core/views/) from mongoose.
+       */
+      createCollection<T extends AnyObject = AnyObject>(name: string, options: mongodb.CreateCollectionOptions, callback: Callback<mongodb.Collection<T>>): void;
+      createCollection<T extends AnyObject = AnyObject>(name: string, callback: Callback<mongodb.Collection<T>>): void;
+      createCollection<T extends AnyObject = AnyObject>(name: string, options?: mongodb.CreateCollectionOptions): Promise<mongodb.Collection<T>>;
+
+      /**
+       * Removes the model named `name` from this connection, if it exists. You can
+       * use this function to clean up any models you created in your tests to
+       * prevent OverwriteModelErrors.
+       */
+      deleteModel(name: string): this;
+
+      /**
+       * Helper for `dropCollection()`. Will delete the given collection, including
+       * all documents and indexes.
+       */
+      dropCollection(collection: string, cb: CallbackWithoutResult): void;
+      dropCollection(collection: string): Promise<void>;
+
+      /**
+       * Helper for `dropDatabase()`. Deletes the given database, including all
+       * collections, documents, and indexes.
+       */
+      dropDatabase(cb: CallbackWithoutResult): void;
+      dropDatabase(): Promise<void>;
+
+      /** Gets the value of the option `key`. */
+      get(key: string): any;
+
+      /**
+       * Returns the [MongoDB driver `MongoClient`](http://mongodb.github.io/node-mongodb-native/3.5/api/MongoClient.html) instance
+       * that this connection uses to talk to MongoDB.
+       */
+      getClient(): mongodb.MongoClient;
+
+      /**
+       * The host name portion of the URI. If multiple hosts, such as a replica set,
+       * this will contain the first host name in the URI
+       */
+      readonly host: string;
+
+      /**
+       * A number identifier for this connection. Used for debugging when
+       * you have [multiple connections](/docs/connections.html#multiple_connections).
+       */
+      readonly id: number;
+
+      /**
+       * A [POJO](https://masteringjs.io/tutorials/fundamentals/pojo) containing
+       * a map from model names to models. Contains all models that have been
+       * added to this connection using [`Connection#model()`](/docs/api/connection.html#connection_Connection-model).
+       */
+      readonly models: Readonly<{ [index: string]: Model<any> }>;
+
+      /** Defines or retrieves a model. */
+      model<T, U, TQueryHelpers = {}>(
+            name: string,
+            schema?: Schema<T, U, TQueryHelpers>,
+            collection?: string,
+            options?: CompileModelOptions
+        ): U;
```

Comment:
ReturnType U seems odd. Should it be Model<U> or does U should extend from Model?

Reply:
`U` should extend from Model, but I believe we removed that for perf reasons in #10349. `U extends Model` causes perf issues for the TS compiler